gridstatus/miso.py

```diff
@@ -679,8 +679,600 @@ def _get_generation_outages_data(
+                "CURVETYPE",
+                "BP1",
+                "PC1",
+                "BP2",
+                "PC2",
+                "BP3",
+                "PC3",
+                "BP4",
+                "PC4",
+                "OVERRIDE",
+                "REASON",
+            ]
+        ]
+
+    @support_date_range(frequency="DAY_START")
+    def get_miso_reserve_product_binding_constraints_day_ahead_hourly(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp = None,
+        verbose: bool = False,
+    ):
+        query_date = date - pd.Timedelta("1D")
+        url = f"https://docs.misoenergy.org/marketreports/{query_date.strftime('%Y%m%d')}_da_rpe.xls"
+        logger.info(
+            f"Downloading day-ahead reserve product binding constraints data from {url}",
+        )
+
+        excel_file = pd.ExcelFile(url)
+        market_date, publish_date = self._get_constraint_header_dates_from_excel(
+            excel_file,
+        )
+        data = pd.read_excel(excel_file, skiprows=3)
+
+        # NOTE(kladar): The last row is a text disclaimer, and there is a leading space
+        # in the column names, so we clean it all up.
+        data = data.iloc[:-1]
+        data["Interval End"] = market_date + pd.to_timedelta(
+            data[
+                "Hour of Occurence"
+            ],  # NOTE(kladar): sic, this is a persistent typo in the header from MISO
+            unit="h",
+        )
+        data["Interval Start"] = data["Interval End"] - pd.Timedelta(hours=1)
+
+        return data[
+            [
+                "Interval Start",
+                "Interval End",
+                "Constraint Name",
+                "Shadow Price",
+                "Constraint Description",
+            ]
+        ]
+
+    @support_date_range(frequency="DAY_START")
+    def get_miso_binding_constraints_real_time_5_min(
+        self,
+        date: str | pd.Timestamp,
+        end: str | pd.Timestamp = None,
+        verbose: bool = False,
+    ):
+        query_date = date + pd.Timedelta("1D")
+        url = f"https://docs.misoenergy.org/marketreports/{query_date.strftime('%Y%m%d')}_rt_bc.xls"
+        logger.info(f"Downloading real-time binding constraints data from {url}")
+
+        excel_file = pd.ExcelFile(url)
+        market_date, publish_date = self._get_constraint_header_dates_from_excel(
+            excel_file,
+        )
+        data = pd.read_excel(
+            excel_file,
+            skiprows=3,
+            dtype={
+                "Constraint Description": object,
+            },
+        )
+
+        data["Interval End"] = pd.to_datetime(
+            market_date.strftime("%Y-%m-%d")
+            + " "
+            + data[
+                "Hour of  Occurrence"
+            ],  # NOTE(kladar): sic, there are two spaces between "Hour of" and "Occurrence"
+        ).dt.tz_localize(
+            self.default_timezone,
+        )  # TODO Confirm that these are hour_ending times
```

Comment:
@cwaldoch , confirming, is the "Hour of Occurrence" in the realtime 5 minute binding constraint interval-ending?